### PR TITLE
Add description on what goes into the next release

### PR DESCRIPTION
**Pull request by potiuk**
---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

**Comment by potiuk** [comment]
This is a follow-up after discussion and question from the user https://github.com/apache/airflow/issues/32928#issuecomment-1783709930 on what can expect to go to the next release. We've been discussing it in the past, here is a proposal to largely describe what is the status-quo.

File: README.md [hunk 1]
```diff
@@ -446,6 +450,42 @@ If you would like to become a maintainer, please review the Apache Airflow
...
+release manager to cherry-pick them is made. If successful, they will be released in `2.7.3`. The final
+decision about cherry-picking is made by the release manager.
+
+Marking issues with a milestone is a bit different. Maintainers do not mark issues with a milestone usually,
+normally they are only marked in PRs. If PR linked to the issue (and "fixing it") gets merged and released
+in a specific version following the process described above, the issue will be automatically closed, no
+milestone will be set for the issue, you need to check the PR that fixed the issue to see which version
+it was released in. However sometimes maintainers mark issues with specific milestone, which means that the
+issue is important to become a candidate to take a look when the release is being prepared. Since this is an
+Open-Source project, where basically all contributors volunteer they time, there is no guarantee that specific
```
Comment (aritra24) [inline README.md]: Nit: their* time

**Review by aritra24 — APPROVED** [review]
Lgtm, just 1 comment for a typo

**Review by pankajkoti — APPROVED** [review]

**Review by ephraimbuddy — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (potiuk) [inline README.md]:
```suggestion
Open-Source project, where basically all contributors volunteer their time, there is no guarantee that specific
```

**Review by pierrejeambrun — COMMENTED** [review]
I think it is important to mention semver somewhere, and maybe link to the spec in case users want more details about core release policy.

**Comment by pierrejeambrun** [comment]
Maybe just link to the Semver section in that same readme ?

**Comment by potiuk** [comment]
> I think it is important to mention semver somewhere, and maybe link to the spec in case users want more details about core release policy.

Yep. Added a short chapter - linking to both SemVer and our "user" documentation about releases.

**Review by jscheffl — APPROVED** [review]

**Review by Lee-W — APPROVED** [review]

**Comment by potiuk** [comment]
Let me merge it so that people can see it - if there are any more corrections, they can be added later :)

**Pull request merged**